### PR TITLE
Feat/drop componentWillReceiveProps

### PR DESCRIPTION
**Pull request by stepancar**
In [7.8.0 eslint-plugin-react release](https://github.com/yannickcr/eslint-plugin-react/releases/tag/v7.8.0) added warning for componentWillRecieveProps https://github.com/yannickcr/eslint-plugin-react/pull/1750
After update package-lock  and reinstall dependencies linter crashes with error.
This pull migrate componentWillRecieveProps to new react lifecicle method getDerivedStateFromProps

### CODE REVIEW

File: src/rsg-components/Playground/Playground.js [hunk 1]
```diff
@@ -30,18 +30,24 @@ export default class Playground extends Component {
...
 
-	componentWillReceiveProps(nextProps) {
+	static getDerivedStateFromProps(nextProps, prevState) {
 		const { code } = nextProps;
-		this.setState({
-			code,
-		});
+		if (prevState.prevCode !== code) {
+			return {
+				...prevState,
```
Comment (sapegin) [inline src/rsg-components/Playground/Playground.js]: You don't need this line, it will merge new state with the previous for you.

**Comment by sapegin** [comment]
And thanks again! 🦄

File: src/rsg-components/Playground/Playground.js (hunk 1 above)
Comment (stepancar) [reply src/rsg-components/Playground/Playground.js]: @sapegin yes, you're right. Thanks

**Comment by stepancar** [comment]
@sapegin added polyfill for support react < 16.4

**Comment by sapegin** [comment]
That's awesome! Thanks!

**Pull request merged**